rpc/internal/test_utils.go

```diff
@@ -0,0 +1,8 @@
+package internal
+
+var globalPortCounter = 3999
+
+func getPort() int {
+	globalPortCounter += 1
+	return globalPortCounter
```

Comment:
There's a pattern that we use in a few places where we do this with a channel and a thread that adds ints and returns ints (which avoids using a lock, which you should probably have regardless, I think. 

Also ++ rather than += 1 

also this file should be named utils_test.go so that it only builds in otherwise this ends up being in prod code.